status/status.go

```diff
@@ -83,8 +85,11 @@ func init() {
 
 // Put adds or replaces status info for the provided name. Any previous setting is returned.
 func Put(name, value string) (previous string, hasPrevious bool) {
+	var rw *sync.RWMutex
```

Comment:
It seems like this will create a new mutex each time this function is called. Multiple goroutines wouldn't share the same lock but instead each have their own locks thus it wouldn't prevent writes to the `info` global var. Since `info` is a global var, I think the lock needs to be a global var as well. A simple unit test where multiple routines try to call `Put` at the same time would be good to add here to help ensure we've covered this case.

It's also better to use `defer` unlock in case there's a panic that happens to ensure that `rw.Unlock` always gets called.

Reply:
I JUST realized this right when you posted this.

Reply:
I am goint to add an unit test here.